parse.go

```diff
@@ -131,16 +131,20 @@ type Context struct {
 type Option func(*Context)
 
 // WithConfigFile tells Parse to read the provided filename as a config file.
-// Requires WithConfigFileParser, and overrides WithConfigFileFlag.
+// Requires WithConfigFileParser, and overrides WithConfigFileFlag. If you
+// want both a config file flag and a default value when one is unprovided,
+// use WithConfigFileFlag and set your chosen default when defining the
+// flag.
 func WithConfigFile(filename string) Option {
 	return func(c *Context) {
 		c.configFile = filename
 	}
 }
 
 // WithConfigFileFlag tells Parse to treat the flag with the given name as a
-// config file. Requires WithConfigFileParser, and is overridden by
-// WithConfigFile.
+// config file. The caller will still need to define the flag in a FlagSet
+// (and can provide a default via that definition). Requires
+// WithConfigFileParser, and is overridden by WithConfigFile.
```

Comment:
I'm not sure how the additional text here provides any additional information. It is pretty clear to me that "treat the flag with the given name as a config file" necessarily implies that "the caller will need to define [that] fiag in [the] FlagSet and can provide a default via that definition".

Reply:
Would it be sufficient to commit the other docs addition but not this one?

Reply:
Or maybe an additional sentence like

> To specify a default config file, provide it as the default value of the corresponding flag — and consider also using the WithAllowMissingConfigFile option.

Reply:
> It is pretty clear to me that "treat the flag with the given name as a config file" necessarily implies that "the caller will need to define [that] fiag in [the] FlagSet and can provide a default via that definition".
I'm coming from a different library that has a similar feature... but defines the config flag for you. In that mindset, I totally expected ff to create its own flag and I wondered what the help text would be.

The proposed sentence is great, I'll swap to that.